hnswlib/bruteforce.h

```diff
@@ -78,7 +78,8 @@ namespace hnswlib {
             dict_external_to_internal.erase(cur_external);
 
             labeltype label=*((labeltype*)(data_ + size_per_element_ * (cur_element_count-1) + data_size_));
-            dict_external_to_internal[label]=cur_c;
+            if (cur_external != label)
```

Comment:
I guess a check for `cur_element_count>1` would be easier to read.
Also I think copying of the element in the next lines do not make any sense so should be in the same `if`